File: app/src/main/java/com/risalfajar/kamusoxford/OxfordApi.java
package com.risalfajar.kamusoxford;

import com.risalfajar.kamusoxford.Entity.SearchResult;
import com.risalfajar.kamusoxford.Entity.Word;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface OxfordApi {

    @GET("search/en")
    Call<SearchResult> searchWord(@Query("q") String keyword, @Header("app_id") String appId, @Header("app_key") String apiKey);

    @GET("entries/en/{id}")
    Call<List<Word>> getWord(@Path("id") String wordId);
}

File: app/src/main/java/com/risalfajar/kamusoxford/service/networkdatasource/OxfordApi.kt
package com.risalfajar.kamusoxford.service.networkdatasource

import com.risalfajar.kamusoxford.service.entity.SearchResult
import com.risalfajar.kamusoxford.service.entity.Word
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path
import retrofit2.http.Query

interface OxfordApi {
    @GET("search/en-us")
    fun searchWord(@Query("q") keyword: String?, @Header("app_id") appId: String?, @Header("app_key") apiKey: String?): Call<SearchResult?>?

    @GET("entries/en-us/{id}")
    fun getWord(@Path("id") wordId: String?): Call<List<Word?>?>?
}